src/game/savegame.c

```diff
@@ -544,3 +544,29 @@ void Savegame_ScanSavedGames(void)
 
     g_SaveCounter++;
 }
+
+void Savegame_ScanAvailableLevels(REQUEST_INFO *req)
+{
+    SAVEGAME_INFO *savegame_info =
+        &m_SavegameInfo[g_InvExtraData[IED_SAVEGAME_NUM]];
+
+    req->items = 0;
+    for (int i = 1; i <= g_GameFlow.last_level_num; i++) {
```

Comment:
I think this could break if the Gameflow has more levels than the requestor slots which I talked about above.

Reply:
Let's use that other trick I mentioned.

LMK when I can review the rest, for now I didn't look at the code yet, just addressed your questions here.